protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -365,23 +368,51 @@ fn execute_cheatcode_hint(
                     nonce,
                     ..tx
                 }));
-            let tx_result = account_tx.execute(blockifier_state, block_context).unwrap();
-            let return_data = tx_result
-                .execute_call_info
-                .expect("Failed to get execution data from method")
-                .execution
-                .retdata;
-            let contract_address = return_data
-                .0
-                .get(0)
-                .expect("Failed to get contract_address from return_data");
-            let contract_address = Felt252::from_bytes_be(contract_address.bytes());
-
-            insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
-            // todo in case of error, consider filling the panic data instead of packing in rust
-            insert_value_to_cellref!(vm, panic_data_start, Felt252::from(0))?;
-            insert_value_to_cellref!(vm, panic_data_end, Felt252::from(0))?;
-
+            match account_tx.execute(blockifier_state, block_context) {
+                Result::Ok(tx_info) => {
+                    let return_data = tx_info
+                        .execute_call_info
+                        .expect("Failed to get execution data from method")
+                        .execution
+                        .retdata;
+                    let contract_address = return_data
+                        .0
+                        .get(0)
+                        .expect("Failed to get contract_address from return_data");
+                    let contract_address = Felt252::from_bytes_be(contract_address.bytes());
+
+                    insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
+                    insert_value_to_cellref!(vm, panic_data_start, Felt252::from(0))?;
+                    insert_value_to_cellref!(vm, panic_data_end, Felt252::from(0))?;
+                }
+                Result::Err(e) => {
+                    if let TransactionExecutionError::ExecutionError(
+                        EntryPointExecutionError::VirtualMachineExecutionErrorWithTrace {
+                            source,
+                            trace,
+                        },
+                    ) = e
+                    {
+                        let msg = source.to_string();
+                        let unparseable_msg = format!("Deploy failed, full error message: {msg}");
+                        let unparseable_msg = unparseable_msg.as_str();
```

Comment:
I think this could be moved to `else` branch. `format!` is somewhat costly, so I wouldn't run it if not necessary.